### ISSUE TITLE
[Question] How to map key press based on dynamic context

### ISSUE THREAD

**Issue by miversen33** [hrsh7th/nvim-cmp#1270]
I have a feeling this is explained somewhere in the docs and I just missed it, so I figure I will ask it here :)

I would like to have my arrow keys (namely, up or down, or k and j) tied to navigating my completion menu selections up or down, but _only_ if I have already entered the completion menu buffer (likely via `tab`). I'm not quite sure how to accomplish this without significantly neovim buffer keymap fuckery.

Basically something like
```lua
cmp.setup({
    mapping = {
        ['<Up>']   = cmp.mapping(cmp.mapping.select_prev_item(), { 'i', 'c' }),
        ['<Down>'] = cmp.mapping(cmp.mapping.select_next_item(), {'i', 'c']})
    }
})
```

But only activating those 2 maps if I have triggered focus on the completion buffer.

**Comment by hrsh7th** [comment]
See https://github.com/hrsh7th/nvim-cmp/blob/main/doc/cmp.txt#L209
